ls1intum/Artemis
# [Refactor] use multiple results for the complaint handling

### Checklist


### Motivation and Context



When a complaint was created previously the original result was simply overridden. To not lose the original assessment, the result was stored in JSON format in the complaint. The two main problems with this are:
1. the result is not easily accessible anymore as it must be parsed before it can be accessed. Until now no mechanism to retrieve this result again did exist -> it is basically write only.
2. the comlaint table is bloated as we store a complete result for each complaint.

We want to solve this by utilizing the multiple results of submissions.

### Description

Now, instead of saving the original assessment as JSON and then overriding it, we just leave it as it is, and add the updated result to the submission's result list. 
Like this both the previous and new result are easily accessible.
<img src=https://user-images.githubusercontent.com/33342534/121494865-6a185d80-c9d9-11eb-8585-a2f14b3c5a35.png width=650>


### Steps for Testing

From a user perspective everything stays mostly the same.

1. Create Exercises (and exam with exercises): FileUpload, Modeling, Text and Programming (note for programming: you should set all dates)
2. Participate with students
3. Assess the submissions with a **Tutor** account
4. Complain with the students (click on the exercise -> click on view to see the result)
5. Now with the same tutor as before check if the complaints are accessible, they should not be!
6. With a different tutor access the complaints from the exercise assessment dashboard
7. Add new feedback, and accept the complaint. (also test rejecting and accepting feedback requests)
8. Now the student should see the updated result.

It should still be possible to open results + complaints which were created before this PR, the existing functionality should not be affected! 

Note for students: for programming exercises the complaint interaction is in the exercise details page on the bottom.

### Test Coverage




Nort changed as only existing and already tested functionality is adapted
